process/block/metablock.go

```diff
@@ -1994,9 +2085,31 @@ func (mp *metaProcessor) computeExistingAndRequestMissingShardHeaders(metaBlock
 		if hdr.GetNonce() > mp.hdrsForCurrBlock.highestHdrNonce[shardData.ShardID] {
 			mp.hdrsForCurrBlock.highestHdrNonce[shardData.ShardID] = hdr.GetNonce()
 		}
+
+		shouldConsiderProofsForNotarization := mp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, hdr.GetEpoch())
+		if shouldConsiderProofsForNotarization {
+			notarizedShardHdrsBasedOnProofs++
+
+			hasProofForShardHdr := mp.proofsPool.HasProof(shardData.ShardID, shardData.HeaderHash)
+			if !hasProofForShardHdr {
+				// TODO: consider verifying the proof before adding it into the proofsPool
+				errAddProof := mp.proofsPool.AddProof(&block.HeaderProof{
+					PubKeysBitmap:       shardData.CurrentPubKeysBitmap,
+					AggregatedSignature: shardData.CurrentSignature,
+					HeaderHash:          shardData.HeaderHash,
+					HeaderEpoch:         hdr.GetEpoch(),
+					HeaderNonce:         shardData.Nonce,
+					HeaderShardId:       shardData.ShardID,
+				})
+				if errAddProof != nil {
+					log.Trace("could not add proof from shard data for header", "hash", hex.EncodeToString(shardData.HeaderHash))
+				}
+			}
+		}
 	}
 
-	if mp.hdrsForCurrBlock.missingHdrs == 0 {
+	shouldRequestMissingFinalityAttestingShardHeaders := notarizedShardHdrsBasedOnProofs != len(metaBlock.ShardInfo)
```

Comment:
check this only if equivalent flag enabled?

Reply:
not needed, here notarizedShardHdrsBasedOnProofs would be 0, thus the second condition will always be true